core/src/main/java/fr/sncf/osrd/envelope/MaxEnvelopeBuilder.java

```diff
@@ -0,0 +1,224 @@
+package fr.sncf.osrd.envelope;
+
+import static fr.sncf.osrd.envelope.MaxEnvelopeBuilder.EnvelopeChangeType.*;
+import static fr.sncf.osrd.envelope.MaxEnvelopeBuilder.EnvelopeChangeType.END;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import java.util.ArrayList;
+import java.util.Collections;
+
+public class MaxEnvelopeBuilder {
```

Comment:
The whole file seems extremely close to `MinEnvelopeBuilder`, it will be a pain to maintain without more factorization

Reply:
True, this is part of the construction margin stuff. that's a big todo

Reply:
it got moved to the construction margin branch